phoenix-core/src/main/java/org/apache/phoenix/compile/DeleteCompiler.java

```diff
@@ -354,7 +354,13 @@ private List<PTable> getClientSideMaintainedIndexes(TableRef tableRef) {
         }
         return Collections.emptyList();
     }
-    
+
+    /**
+     * Implementation of MutationPlan that is selected if
+     * 1) there is no index presented for a table,
```

Comment:
For mutable indexes it is possible to use this plan.

Reply:
Even immutable indexes can use this plan if the indexed columns can be evaluated from the query

Reply:
Oh yeah, you are right. I will update that.